awx/api/serializers.py

```diff
@@ -390,7 +391,20 @@ def get_related(self, obj):
         view = self.context.get('view', None)
         if view and (hasattr(view, 'retrieve') or view.request.method == 'POST') and type(obj) in settings.NAMED_URL_GRAPH:
             original_url = self.get_url(obj)
-            res['named_url'] = self._generate_named_url(original_url, obj, settings.NAMED_URL_GRAPH[type(obj)])
+
+            # If the app is running at a location other than /, temporarily remove the
+            # prefix. This is to avoid changing the code in _generate_named_url where
+            # it is assumed the ID of the resource is at url_units[4].
```

Comment:
this seems like a hacky way to solve the problem, as opposed to modifying `_generate_named_url` itself.

Reply:
Suggestions? We don't have access to the request object there. Sending the index in as an argument seems equally weird to me.

Reply:
@AlanCoding Please take a look at my latest commit.